tools/diffscripts/diffapi.ps1

```diff
@@ -0,0 +1,203 @@
+# Copyright (c) Microsoft. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+<#
+
+.SYNOPSIS
+Microsoft Azure IoT SDK Release comparison script
+
+.DESCRIPTION
+This script is used to compare the current candidate for release to the previous version that is in the iot-sdk-internals 
+
+Parameters:
+    -AsmToolPath: The path to the compiled AsmDiff tool found in the dotnet arcade (ex: c:\tools\asmdifftool)
+    -SDKInternalsPath: The path of the iot-sdk-internals repository (ex: c:\repo\iot-sdks-internals)
+	-Preview indicates you will compare the output to the last preview version
+
+.EXAMPLE
+.\diffapi
+
+Executes the commands assuming everything is cloned relative to the SDK repo and the AsmDiff tool is compiled using the arcade instructions.
+.EXAMPLE
+.\diffapi -AsmToolPath c:\tools\asmdifftool
+
+Specifies the location of the AsmDiff tool executable to run the diff commands.
+.EXAMPLE
+.\diffapi -SDKInternalsPath c:\repo\iot-sdks-internals
+
+Specifies the location of the SDK Internals repository if not cloned relative to the SDK repository.
+.EXAMPLE
+.\diffapi -Preview
+
+Executes the commands assuming everything is cloned relative to the SDK repo but will compare against the preview versions of the documents.
+.LINK
+https://github.com/azure/azure-iot-sdk-csharp
+
+#>
+
+Param(
+    [ValidateScript({
+            if( -Not ($_ | Test-Path -PathType Container) ){
+                throw "Folder $_ does not exist."
+            }
+            return $true
+        })]
+    # The path to the compiled AsmDiff tool found in the dotnet arcade (ex: c:\tools\asmdifftool)
+    [System.IO.FileInfo] $AsmToolPath = $null,
+    [ValidateScript({
+            if( -Not ($_ | Test-Path -PathType Container) ){
+                throw "Folder $_ does not exist."
+            }
+            return $true
+        })]
+    # The path of the iot-sdk-internals repository (ex: c:\repo\iot-sdks-internals)
+    [System.IO.FileInfo] $SDKInternalsPath = $null,
+    # Indicates you will compare the output to the last preview version instead of master
+    [switch] $Preview
+)
+
+
+$reporoot = (gi $pwd).Parent.Parent.FullName
+$baseroot = (gi $reporoot).Parent.FullName
+
+$asmtoolroot = (Join-Path -Path $baseroot -Child "\arcade\artifacts\bin\Microsoft.DotNet.AsmDiff\Debug\netcoreapp3.1\") 
+if ($AsmToolPath -ne $null) {
```

Comment:
What if it is null?

Reply:
If it's null it will just use the $asmToolRoot specified above.